### PR TITLE
Refactored unused vars to _

### LINKED ISSUE #9855: Code quality refactoring
I was doing a code quality analysis with [SonarQube](www.sonarqube.org) on Bokeh and came through several refactoring candidates that may be addressed quickly and I believe would benefit the overall project. Some of them may be minor bugs but still.
Here are the main issues (that may be repeated througout the code)

### Readability issues

- Nested ```if``` statements that may be merged:

```python
if condition1:
    if condition2:
        # ...
```
to 
```python
if condition1 and condition2:
        # ...
```

### Maintainability Issues

- Repeated constants should be assigned to variable for easier future refactoring. 

### Efficiency

- In cycles (for/while) the use of if/elif and continue/break to improve performance and excecution time when its not necesary to excecute the remaining code

### Cognitive complexity

- Some functions are very cognitively complex, some of them could be eithermodularized or simplified. Some of the issues mentioned beforehand even reduce complexity in some of them.

### Repeated turnouts in conditions

[plots.py](https://github.com/bokeh/bokeh/blob/master/bokeh/models/plots.py#L357) has the same result for the if and elif conditions. It can be replaced by the ```or``` operator as its more efficient and clear. (or stops checking the remaining conditions once it finds a ```True```) run this code if you dont believe me:
```python
def f():
	print("text")
	return False
print(True or f())
```

### switch case issues

- Some switch functions are missing the break after each ```case``` test, making the switch statement lose its puropose. (both cases are completely different) 
- ~Also, when evaluating ```case a || b```, only ```a``` is considered. Should be wrapped around parenthesis or considered as two separate cases [templating.js line 35](https://github.com/bokeh/bokeh/blob/master/bokehjs/src/lib/core/util/templating.ts#L35)~

### indent and curly braces in if/else {}

This is just a good practice and a very minor issue, but it improves readability for future maintainers

### Missing type in [```palettes.ts```](https://github.com/bokeh/bokeh/blob/master/bokehjs/src/lib/api/palettes.ts)
- Set3 considers Set3_11 twice and doesnt consider Set3_12

####  length >= 0
- In [index_filters.ts](https://github.com/bokeh/bokeh/blob/master/bokehjs/src/lib/models/filters/index_filter.ts) line 32 an array length is checked to be greater **or equal** than 0... Is this a mistake or on purpose?

Is it alright if I fix these issues and submit a PR? If any of them are intentional for some reason, please let me know :+1:

### PR DESCRIPTION
**Pull request by auirarrazaval**
addresses #9855

### CODE REVIEW

File: bokeh/_version.py [hunk 1]
```diff
@@ -500,7 +500,7 @@ def get_versions():
         # versionfile_source is the relative path from the top of the source
         # tree (where the .git directory might live) to this file. Invert
         # this to find the root from __file__.
-        for i in cfg.versionfile_source.split('/'):  # lgtm [py/unused-loop-variable]
+        for _ in cfg.versionfile_source.split('/'):  # lgtm [py/unused-loop-variable]
```
Comment (mattpap) [inline bokeh/_version.py:503]: I suppose the `lgtm` comment can be removed now. It's interesting that lgtm complained here and not in the other cases. I'm wondering how reliable is that tool in general.

Comment (auirarrazaval) [reply bokeh/_version.py:503]: Yeah i did think about that as well. Also, I don't get why it was used here in the first place

**Comment by bryevdv** [comment]
Re: LGTM no tool is perfect so the question becomes one of usefulness. Having spent time with this now I would say the rate of useful/valid things fixed to "false positives" is easily greater than 10 to 1, which is certainly high enough for me.

File: bokeh/_version.py (hunk 1 above)
Comment (bryevdv) [reply bokeh/_version.py:503]: @auirarrazaval can you remove the lgtm comment here so that this can be merged?

Comment (auirarrazaval) [reply bokeh/_version.py]: @bryevdv  Done

**Comment by bryevdv** [comment]
Thanks @auirarrazaval

**Pull request merged**